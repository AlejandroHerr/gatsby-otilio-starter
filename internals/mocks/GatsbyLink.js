/* eslint-disable */
import React from 'react';

const GatsbyLink = ({ children, className, to }) => (
  <a className={className} href={to}>
    {children}
  </a>
);

export default GatsbyLink;
