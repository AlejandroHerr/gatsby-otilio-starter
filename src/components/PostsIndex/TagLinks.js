// @flow
import React from 'react';
import Link from 'gatsby-link';

type PropsType = {
  tags: Array<string>,
};

const TagLinks = ({ tags }: PropsType) => (
  <span>{' on '}{tags.map((tag, idx) => (idx < (tags.length - 1)
  ? (
    <span key={tag}>
      <Link to={`/tags/${tag}`}>{tag}</Link>{', '}
    </span>
  )
  : (
    <span key={tag}>
      <Link to={`/tags/${tag}`}>{tag}</Link>
    </span>
  )))}
  </span>
);

export default TagLinks;
