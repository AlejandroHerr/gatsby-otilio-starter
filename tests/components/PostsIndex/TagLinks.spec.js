import React from 'react';
import { shallow } from 'enzyme';
import GatsbyLink from 'gatsby-link';

import TagLinks from '../../../src/components/PostsIndex/TagLinks';

const defaultProps = {
  tags: ['tag1', 'tag2', 'tag3'],
};

const setup = (props = {}) => {
  const tagLinks = shallow(<TagLinks {...defaultProps} {...props} />);

  return {
    tagLinks,
    link: tagLinks.find(GatsbyLink),
  };
};

describe('PostsIndex', () => {
  describe('TagLinks', () => {
    it('should render a link to the tag page for every tag', () => {
      const { tagLinks, link } = setup();


      expect(link).toHaveLength(defaultProps.tags.length);
      defaultProps.tags.forEach((tag, idx) => {
        expect(link.at(idx).prop('to')).toBe(`tag/${tag}`);
        expect(link.at(idx).prop('children')).toBe(tag);
      });
      expect(tagLinks).toMatchSnapshot();
    });
  });
});
