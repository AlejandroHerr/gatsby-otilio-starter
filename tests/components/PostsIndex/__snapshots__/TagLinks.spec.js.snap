// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsIndex TagLinks should render a link to the tag page for every tag 1`] = `
<span>
   on 
  <span
    key="tag1"
  >
    <GatsbyLink
      to="/tags/tag1"
    >
      tag1
    </GatsbyLink>
    , 
  </span>
  <span
    key="tag2"
  >
    <GatsbyLink
      to="/tags/tag2"
    >
      tag2
    </GatsbyLink>
    , 
  </span>
  <span
    key="tag3"
  >
    <GatsbyLink
      to="/tags/tag3"
    >
      tag3
    </GatsbyLink>
  </span>
</span>
`;
